src/celeritas/global/Stepper.hh

```diff
@@ -191,5 +191,12 @@ class Stepper final : public StepperInterface
     std::shared_ptr<ActionSequenceT> actions_;
 };
 
+//---------------------------------------------------------------------------//
+// EXPLICIT INSTANTIATION
+//---------------------------------------------------------------------------//
+
+extern template class Stepper<MemSpace::host>;
+extern template class Stepper<MemSpace::device>;
```

Comment:
This class has a public non-virtual dtor.

Reply:
It's `final` so it doesn't need to be explicitly virtual? The base class is protected.

Reply:
But there's no good reason not to make the base destructor virtual so I'll do that.

Reply:
> It's final so it doesn't need to be explicitly virtual?

`Stepper`'s destructor has apriori no reason to be more virtual than it's base class.  However since the base `StepperInterface` exists, users could have tried to call the operator delete on a pointer to the base class ... except for the fact that that the destructor of the base class was declared protected ... but this prevents only external calls to the base class operator delete but does not prevent calls to that same operator from any of the `Stepper` classes ... and this misuse would not be caught with the current code ... to prevent it we would need to declare the operator delete itself private (we can't declare the destructor private since it is needed by the derived classes).

TLDR:  Seth is right, just make the base class destructor virtual since it already has a virtual table.

Reply:
@pcanal If anyone tries to submit a PR that contains `delete this`, they're off the project 😂

Reply:
This could happens a little bit easier 
```
class Base { protected: ~Base(); };
class Derived { Base *other; public: ~Derived() { delete other; } };
```

Reply:
Ha, good point. That (and any other use of `delete` of Celeritas objects) would only merit a stern talking-to 😉